dhis-2/dhis-api/src/main/java/org/hisp/dhis/reporttable/ReportTable.java

```diff
@@ -387,21 +393,35 @@ public void populateGridColumnsAndRows( Date date, User user,
     }
 
     @Override
-    public void populateAnalyticalProperties()
-    {
-        for ( String column : columnDimensions )
+    public void populateAnalyticalProperties() {

```

Comment:
Added some preventive code in case the database data/rows are inconsistent, ie: contains wrong sort_order or no columns/rows at all (depending on the Visualization)